internal/users/manager.go

```diff
@@ -237,6 +304,59 @@ func (m *Manager) UpdateUser(u types.UserInfo) (err error) {
 	return nil
 }
 
+func (m *Manager) getCurrentUserFromDB(name string) (currentUser *types.UserInfo, err error) {
```

Comment:
I don't love that we're using the `types.UserInfo` type here, which is otherwise only used as a return value from the broker to authd. IMO it's confusing that we're now also using it internally in the `users` package, in addition to `types.UserEntry`. However, I get why it's useful, so I'm fine with keeping it if we make the distinction clear in the function and variable names.

```suggestion
func (m *Manager) getUserInfo(name string) (userInfo *types.UserInfo, err error) {
```

Reply:
If you like I can also make a type alias, so just use `users.User....Someting`?  :)